internal/metrics/FileStorage.go

```diff
@@ -0,0 +1,124 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"gmetrics/internal/contextkeys"
+	"gmetrics/internal/logger"
+	"gmetrics/internal/metrics/fileworker"
+	"io"
+	"os"
+	"time"
+)
+
+// Writer интерфейс записывающего в файл типа
+type Writer interface {
+	Write(v any) error
+	Close() error
+}
```

Comment:
Здесь можно попробовать использовать комбинацию интерфейсов и исользовать уже существующий интерфейс из пакета `io`.

```suggestion
type Writer interface {
	Write(v any) error
	io.Closer
}
```

Reply:
Добавил интерфейс io.Closer